lib/galaxy/objectstore/onedata.py

```diff
@@ -0,0 +1,557 @@
+        verify_ssl = string_as_bool(conn_xml.get("verify_ssl", "True"))
+
+        space_xml = _get_config_xml_elements(config_xml, "space")[0]
+        space_name = space_xml.get("name")
+
+        cache_dict = parse_caching_config_dict_from_xml(config_xml)
+
+        extra_dirs = [{attr: elem.get(attr) for attr in ("type", "path")} 
+                      for elem in _get_config_xml_elements(config_xml, "extra_dir")]
+
+        return {
+            "auth": {
+                "access_token": access_token
+            },
+            "connection": {
+                "onezone_domain": onezone_domain,
+                "verify_ssl": verify_ssl,
+            },
+            "space": {
+                "name": space_name
+            },
+            "cache": cache_dict,
+            "extra_dirs": extra_dirs,
+            "private": ConcreteObjectStore.parse_private_from_config_xml(config_xml),
+        }
+    except Exception:
+        # Toss it back up after logging, we can't continue loading at this point.
+        log.exception("Malformed ObjectStore Configuration XML -- unable to continue")
+        raise
+
+
+def _get_config_xml_elements(config_xml, tag):
+    elements = config_xml.findall(tag)
+
+    if not elements:
+        msg = f"No {tag} element in config XML tree"
+        log.error(msg)
+        raise Exception(msg)
+
+    return elements
+
+
+class OnedataObjectStore(ConcreteObjectStore):
+    """
+    Object store that stores objects as items in an Onedata. A local
+    cache exists that is used as an intermediate location for files between
+    Galaxy and Onedata.
+    """
+
+    cache_monitor: Optional[InProcessCacheMonitor] = None
+    store_type = "onedata"
+
+    def __init__(self, config, config_dict):
+        super().__init__(config, config_dict)
+        self.cache_monitor = None
+
+        auth_dict = config_dict["auth"]
+        self.access_token = auth_dict["access_token"]
+
+        connection_dict = config_dict["connection"]
+        self.onezone_domain = connection_dict["onezone_domain"]
+        self.verify_ssl = connection_dict.get("verify_ssl", True)
+
+        space_dict = config_dict["space"]
+        self.space_name = space_dict["name"] 
+
+        cache_dict = config_dict.get("cache") or {}
+        self.enable_cache_monitor, self.cache_monitor_interval = enable_cache_monitor(config, config_dict)
+        self.cache_size = cache_dict["size"] or self.config.object_store_cache_size
+        self.staging_path = cache_dict["path"] or self.config.object_store_cache_path
+        self.cache_updated_data = cache_dict.get("cache_updated_data", True)
+
+        extra_dirs = {e["type"]: e["path"] for e in config_dict.get("extra_dirs", [])}
+        self.extra_dirs.update(extra_dirs)
+
+        self._initialize()
+
+    def _initialize(self):
+        if OnedataFileRESTClient is None:
+            raise Exception(NO_ONEDATA_ERROR_MESSAGE)
+
+        log.debug("Configuring Onedata Connection")
```

Comment:
może `log.debug("Configuring Onedata connection to {onezone_domain} (verify_ssl={verify_ssl})")`